src/components/Quiz/Quiz.tsx

```diff
@@ -0,0 +1,196 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+import "./Quiz.css";
+import { useQuiz } from "../../contexts/Quiz-Context";
+import { Modals } from "../Modal/Modal";
+import { Timer } from "../Timer/Timer";
+import { useParams } from "react-router-dom";
+import React from "react";
+import { createStyles, Theme, makeStyles } from "@material-ui/core/styles";
+import Chip from "@material-ui/core/Chip";
+import BubbleChartIcon from "@material-ui/icons/BubbleChart";
+import Button from "@material-ui/core/Button";
+import Loader from "react-loader-spinner";
+
+export function Quiz() {
+  const { id } = useParams();
+  const { state, dispatch } = useQuiz();
+
+  useEffect(() => {
+    (async function () {
+      try {
+        const response = await axios.get(
+          `https://Quiz-App-API.sayuk.repl.co/${id}`
+        );
+        console.log(response.data, "ressss");
+
+        dispatch({ type: "SET_DATA", payload: { data: response.data } });
+      } catch (err) {
+        console.log(err, "errr");
+      }
+    })();
+  }, []);
+
+  console.log(state.data.questions);
+
+  const useChip = makeStyles((theme: Theme) =>
+    createStyles({
+      root: {
+        display: "flex",
+        justifyContent: "center",
+        flexWrap: "wrap",
+        "& > *": {
+          margin: theme.spacing(0.5),
+          // fontWeight: "bolder",
+          alignItems: "center",
+          color: "white",
+        },
+      },
+    })
+  );
+
+  const useButton = makeStyles((theme: Theme) =>
+    createStyles({
+      root: {
+        "& > *": {
+          margin: theme.spacing(1),
+          color: "white",
+          padding: "0rem",
+        },
+      },
+    })
+  );
+
+  const useSecButton = makeStyles((theme: Theme) =>
+    createStyles({
+      root: {
+        "& > *": {
+          margin: theme.spacing(1),
+          color: "color",
+        },
+      },
+    })
+  );
+
+  const chip = useChip();
+  const button = useButton();
+  const secBtn = useSecButton();
+
+  const [selected, setSelected] = useState("");
+
+  console.log(selected, "sell");
+
+  const checkHandler = (selectedAns: any) => {
```

Comment:
Avoid `any` and give appropriate types

Reply:
Done that.
Thanks, improves the code quality to quite some extent.